cns/restserver/restserver.go

```diff
@@ -19,19 +20,27 @@ import (
 	"github.com/Azure/azure-container-networking/cns/imdsclient"
 	"github.com/Azure/azure-container-networking/cns/ipamclient"
 	"github.com/Azure/azure-container-networking/cns/networkcontainers"
+	"github.com/Azure/azure-container-networking/cns/nmagentclient"
 	"github.com/Azure/azure-container-networking/cns/routes"
 	acn "github.com/Azure/azure-container-networking/common"
 	"github.com/Azure/azure-container-networking/log"
 	"github.com/Azure/azure-container-networking/platform"
 	"github.com/Azure/azure-container-networking/store"
 )
 
+var (
+	// Named Lock for accessing different states in httpRestServiceState
+	namedLock = acn.InitNamedLock()
+)
+
 const (
 	// Key against which CNS state is persisted.
 	storeKey        = "ContainerNetworkService"
 	swiftAPIVersion = "1"
 	attach          = "Attach"
 	detach          = "Detach"
+	// Rest service state identifier for named lock
+	stateJoinedNetworks = "JoinedNetworks"
```

Comment:
are these persisted?

Reply:
not persistent.

Reply:
this is just a identifier for the lock. I don't persist the joined networks